content/articles/mongo-nodejs-cubejs-dashboard/index.md

```diff
@@ -0,0 +1,249 @@
+- [Prerequisites](#prerequisites)
+- [Objectives](#objectives)
+- [Setting up the development environment](#setting-up-the-development-environment)
+- [Generating Cube.js schema](#generating-cubejs-schema)
+- [Adding Cube.js environmental variables](#adding-cubejs-environmental-variables)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+
+- This tutorial requires a basic understanding of the MongoDB and MongoDB BI connector.
+- Basic knowledge in Node.js, Express, and Cube.js.
+
+### Objectives
+
+By the end of this tutorial, you should be able to create a simple Node.js application dashboard for analytics, combining with the power of Cube.js and MongoDB.
+
+### Setting up the development environment
+
+Let's begin by installing all the above requirements before starting the development process.  
+
+**Step 1:**
+Download [Node.js](https://nodejs.org/en/download/current/) for your local machine. In this tutorial, we'll be using Node version `v16.5.0` on Ubuntu 20.04. However, it's important to note that the concepts will remain the same for any operating system.  
+
+Check your installed version by running the following command:  
+
+```bash
+node --version
+```
+
+Output:
+
+```bash
+v16.5.0 # note that this version may vary from yours
+```
+
+**Step 2:**  
+Let's now proceed and create our application skeleton using [Express](https://expressjs.com) by running the following commands:
+
+```bash
+npm install express-generator -g
+```
+
+Now proceed and generate our `example-analytics-dashboard` app by executing the following commands:
+
+```bash
+express --view=hbs example-analytics-dashboard # notice that the view is set to use hbs
+```
+
+Expected Output:
+
+```bash
+   create : example-analytics-dashboard/
+   create : example-analytics-dashboard/public/
+   create : example-analytics-dashboard/public/javascripts/
+   create : example-analytics-dashboard/public/images/
+   create : example-analytics-dashboard/public/stylesheets/
+   create : example-analytics-dashboard/public/stylesheets/style.css
+   create : example-analytics-dashboard/routes/
+   create : example-analytics-dashboard/routes/index.js
+   create : example-analytics-dashboard/routes/users.js
+   create : example-analytics-dashboard/views/
+   create : example-analytics-dashboard/views/error.hbs
+   create : example-analytics-dashboard/views/index.hbs
+   create : example-analytics-dashboard/views/layout.hbs
+   create : example-analytics-dashboard/app.js
+   create : example-analytics-dashboard/package.json
+   create : example-analytics-dashboard/bin/
+   create : example-analytics-dashboard/bin/www
+
+   change directory:
+     $ cd example-analytics-dashboard
+
+   install dependencies:
+     $ npm install
+
+   run the app:
+     $ DEBUG=example-analytics-dashboard:* npm start
+
+```
+
+**Step 3:**
+Let's now install the local MongoDB instance and MongoDB BI connector.  
+Begin by running the following command to import the MongoDB instance public key:
```

Comment:
@bhanjibrilliant, imagine of a reader `hearing of the term 'BI connector'` for the first time, what would they do? make your article flow as much as possible. Do not let your readers leave your article. Kindly make this change.